src/App/Fossa/Analyze.hs

```diff
@@ -371,6 +381,27 @@ filterProjects rootDir = filter (isProductionPath . dropPrefix rootPath . fromAb
     dropPrefix :: String -> String -> String
     dropPrefix prefix str = fromMaybe prefix (stripPrefix prefix str)
 
+-- We want to treat IAT dependencies separately so that we can resolve them into user-defined custom dependencies.
+-- Remove any deps from the graph of type IATType.
+-- Evaluates to the list of dependencies so removed and the list of project results after removal.
+extractIATDeps :: [ProjectResult] -> ([Dependency], [ProjectResult])
+extractIATDeps results = do
+  -- hack: process the results twice to get the two pieces of infomation desired.
+  -- I'm happy to do this a better way but I'm not sure how to combine these into one action.
+  let iatDepGraphs = Graphing.filter isIATDep <$> (projectResultGraph <$> results)
+  let iatOnlyDeps = concat $ Graphing.toList <$> iatDepGraphs
+  (iatOnlyDeps, filterIATDeps <$> results)
+  where
+    isIATDep :: Dependency -> Bool
+    isIATDep d = (dependencyType d) == IATType
+    filterIATDeps :: ProjectResult -> ProjectResult
+    filterIATDeps ProjectResult{..} =
+      ProjectResult
+        projectResultType
+        projectResultPath
+        (Graphing.filter (not . isIATDep) projectResultGraph)
+        projectResultGraphBreadth
+
```

Comment:
There's quite a few things I'd like to do about this block.  I don't know if we can prevent processing twice with our filter, but we may be able to do a few other fixes.

* Primarily, we're `map`-ping, and therefore iterating, multiple times, we can fix that.
* We can use record update syntax instead of manually creating another `ProjectResult`.
* We can avoid using `let` and `where` together, which can be confusing.
* We don't need do notation here.

```suggestion
-- FIXME: we do Graphing.filter twice, but one is inverted.  Can we just split the Graphing?
extractIATDeps :: [ProjectResult] -> ([Dependency], [ProjectResult])
extractIATDeps results = (iatOnlyDeps, map filterIATDeps results)
  where
    iatOnlyDeps :: [Dependency]
    iatOnlyDeps = concatMap pullDeps results

    pullDeps :: ProjectResult -> [Dependency]
    pullDeps = Graphing.toList . Graphing.filter isIATDep . projectResultGraph

    isIATDep :: Dependency -> Bool
    isIATDep d = (dependencyType d) == IATType

    filterIATDeps :: ProjectResult -> ProjectResult
    filterIATDeps result = result {projectResultGraph = Graphing.filter (not . isIATDep) $ projectResultGraph result}

```

One piece of advice from the style guide (#325): use `map` instead of `fmap` or `<$>`, since lists are different in performance than a lot of other functors.  Prefer explicit operations on all collection-like functors (or functor-like types, e.g.: `Set`).  Let me know if anything here doesn't make sense, or if it's not clear how we got here from the original code.